src/Lean/Util/FindExpr.lean

```diff
@@ -56,6 +56,18 @@ def find? (p : Expr → Bool) (e : Expr) : Option Expr :=
 def occurs (e : Expr) (t : Expr) : Bool :=
   (t.find? fun s => s == e).isSome
 
+def findM? [Monad m] (p : Expr → m Bool) (e : Expr) : m (Option Expr) := do
```

Comment:
I'm hesitant to merge this as I'm not sure whether this should also receive a `PtrSet`-backed efficient implementation.

Reply:
Once #3970 lands, could you update this PR and ping me again?

Reply:
@semorrison The PR is now rebased to `master`.